app/src/ui/multi-commit-operation/choose-branch/merge-choose-branch-dialog.tsx

```diff
@@ -104,6 +104,12 @@ export class MergeChooseBranchDialog extends BaseChooseBranchDialog {
         return { kind: ComputedAction.Clean }
       })
 
+      if (this.state.selectedBranch !== branch) {
```

Comment:
Should we do this in the `catch` part above too?

Or even more comprehensively… store the result of `await promiseWithMinimumTimeout` in a variable, then do this `selectedBranch != branch` check, and then store it in `this.mergeStatus` to prevent other issues.

Does that make sense? 🤔

Reply:
Because it all sets the `mergeStatus`, I don't think there is needed difference for the `catch`. But.. I do think, you are right and it doesn't make sense to set `this.mergeStatus` until until after the `this.state.selectedBranch` check. I will take another look at it.